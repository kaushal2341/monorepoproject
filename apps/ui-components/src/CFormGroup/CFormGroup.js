import React from 'react';
import {Form} from "react-bootstrap";
import PropTypes from 'prop-types';

const CFormGroup = props => {
    const {
        _ref,
        as,
        bsPrefix,
        children,
        className,
        controlId,
        id
    } = props;
    return (
        <Form.Group
            _ref={_ref}
            as={as}
            bsPrefix={bsPrefix}
            children={children}
            className={className}
            controlId={controlId}
            id={"group".concat("_").concat(id)}
        />)
};

CFormGroup.propTypes = {
    _ref: PropTypes.oneOfType([
        PropTypes.func, // for legacy refs
        PropTypes.shape({current: PropTypes.instanceOf(Element)})
    ]),
    as: PropTypes.elementType,
    bsPrefix: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    controlId: PropTypes.string,
    id: PropTypes.string
};

export default CFormGroup;