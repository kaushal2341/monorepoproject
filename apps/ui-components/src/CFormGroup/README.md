# CFormGroup 

Form.Group component made using react-bootstrap's `Form.Group`.

To use it, first

```jsx harmony
import {CFormGroup} from '@cogent/ui-components';
```
Then simply use it,

```jsx harmony
 let form = (<>
        <input type="text" required/>
        <CButton id="c-button-2" type="submit" name="Submit"/>
   </>);

  <CFormGroup
         id="test"
         controlId="formHorizontalEmail"
         children={form}/>;
```
You can pass in the form components through children props and submit handler function through onSubmit props.

## CFormGroup props:

```text
_ ref: ReactRef - The FormGroup ref will be forwarded to the underlying element. Unless the FormGroup is rendered as a composite component, it will be a DOM node, when resolved.
as: elementType - You can use a custom element type for this component.
bsPrefix: string - defaultValue: 'form-group' - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.
children: node - Children of CFormGroup Component.
className: string - Css classname for additional styling.
controlId: string - Sets id on <FormControl> and htmlFor on <FormGroup.Label>. 
                    If the children component have their own id, they take their own id by default
id: string - id of the FormGroup component.
```
