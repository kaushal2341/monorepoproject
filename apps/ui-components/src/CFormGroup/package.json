{
  "name": "@cogent/c-formgroup",
  "version": "0.0.0",
  "main": "CFormGroup.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
