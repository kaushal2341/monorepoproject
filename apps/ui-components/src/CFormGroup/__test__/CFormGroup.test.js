import CFormGroup from "../CFormGroup";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CFormGroup Component Tests', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CFormGroup id="test"/>);
    });

    test('if CFormGroup Component is defined', () => {
        expect(wrapper.find('CFormGroup')).toBeDefined();
    });

    test('if Form.Group component is defined', () => {
        expect(wrapper.find('#group_test').length).toBe(1);
    });

    test('if all props required for Form.Group component are defined', () => {
        let propRequired = [
            '_ref',
            'as',
            'bsPrefix',
            'children',
            'className',
            'controlId',
            'id'
        ];

        let propsOfFormGroup = Object.keys(wrapper.find('#group_test').props());
        propRequired.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfFormGroup[i])
        ));
    });

    describe('Snapshot Testing', () => {
        test('if matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});