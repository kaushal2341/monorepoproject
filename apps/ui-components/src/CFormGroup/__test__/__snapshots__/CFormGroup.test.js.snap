// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormGroup Component Tests Snapshot Testing if matches snapshot 1`] = `
<FormGroup
  id="group_test"
/>
`;
